ItHappened/Application/StatisticsService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ItHappened.Domain;
+using ItHappened.Domain.Exceptions;
+using ItHappened.Domain.Repositories;
+
+namespace ItHappened.Application
+{
+    public class StatisticsService
+    {
+        public StatisticsService(IEventRepository eventRepository, 
+            ItHappenedSettings settings)
+        {
+            _eventRepository = eventRepository;
+            _settings = settings;
+        }
+
+        private readonly IEventRepository _eventRepository;
+        private readonly ItHappenedSettings _settings;
+
+        public EventDto GetBestEvent(Guid trackId)
+        {
+            Event bestEvent = null;
+            var trackEvents = _eventRepository.TryGetEventsByTrack(trackId);
+            var trackEventsWithRating = trackEvents
+                .Where(@event => @event.Customization.Rating != null).ToList();
+
+            if (trackEventsWithRating.Count() >= _settings.EventsWithRatingCountForSearchingBestEvent
+                && DateTime.Compare(
+                    trackEventsWithRating
```

Comment:
не понимаю, зачем работать с trackEventsWithRating, если можно просто взять первый из отсортированного по дате списка событий. 

сейчас написано так, что пропустив 9 ближайших событий мы не получим раннюю дату. Если события с рейтингом будут происходить несколько раз в день, например на протяжении 6 месяцев, то это удовлетворит условие, но код не сработает. 

> самое раннее событие было раньше, чем 3 месяца назад,

```suggestion
                    trackEvents..OrderBy(@event => @event.CreatedAt).First().CreatedAt
```